File: lintcode/69_binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/binary-tree-level-order-traversal
@Language: C++
@Datetime: 16-05-27 02:08
*/

/**
 * Definition of TreeNode:
 * class TreeNode {
 * public:
 *     int val;
 *     TreeNode *left, *right;
 *     TreeNode(int val) {
 *         this->val = val;
 *         this->left = this->right = NULL;
 *     }
 * }
 */
 
 
class Solution {
    /**
     * @param root: The root of binary tree.
     * @return: Level order a list of lists of integer
     */
public:
    vector<vector<int>> levelOrder(TreeNode *root) {
        vector<vector<int>> res;
        if (!root) return res;
        vector<int> one;
        queue<TreeNode*> que;
        que.push(root);
        int size = 1;
        while(que.size()) {
            TreeNode* cur = que.front();
            que.pop(); size--;
            one.push_back(cur->val);
            if (cur->left) que.push(cur->left);
            if (cur->right) que.push(cur->right);
            if (size == 0) {
                size = que.size();
                res.push_back(one);
                one.clear();
            }
        }
        return res;
    }
};

File: lintcode/69_binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/binary-tree-level-order-traversal
@Language: Python
@Datetime: 16-05-04 00:18
'''

"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""


class Solution:
    """
    @param root: The root of binary tree.
    @return: Level order in a list of lists of integers
    """
    def levelOrder(self, root):
        if not root:
            return []
        que = [root]
        res = []
        lvl = []
        cnt = 1
        while que:
            cur = que.pop(0)
            cnt -= 1
            lvl.append(cur.val)
            if cur.left:
                que.append(cur.left)
            if cur.right:
                que.append(cur.right)
            if cnt == 0:
                cnt = len(que)
                res.append(lvl[:])
                lvl[:] = []
        return res